lib/plugins/aws/customResources/generateZip.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const os = require('os');
+const path = require('path');
+const { memoize } = require('lodash');
+const BbPromise = require('bluebird');
+const childProcess = BbPromise.promisifyAll(require('child_process'));
+const fse = BbPromise.promisifyAll(require('fs-extra'));
+const { version } = require('../../../../package');
+const getTmpDirPath = require('../../../utils/fs/getTmpDirPath');
+const createZipFile = require('../../../utils/fs/createZipFile');
+
+const srcDirPath = path.join(__dirname, 'resources');
+const cachedZipFilePath = path.join(
+  os.homedir(),
+  '.serverless/cache/custom-resources',
```

Comment:
AFAIR we had a cache for the `tabtab` autocomplete as well. Is this the same `cache` directory (couldn't find the code, but AFAIK @medikoo has touched that piece of code recently).

Would be nice if we define and re-use one cache directory for all the stuff we cache throughout the codebase.

Reply:
> Would be nice if we define and re-use one cache directory for all the stuff we cache throughout the codebase.

Yes it's exactly the goal. Currently tabtab uses `.serverless/cache` directly, but it uses uuid hashes - so there's no collisions. With other PR, I'll shift it to `.serverless/cache/tabtab` (and probably it'll be good introduce a command as `serverless cache clean`)

Reply:
Great! Thanks for the clarification :+1: